tests/monitoring_processes/test_monitoring_critical_processes.py

```diff
@@ -0,0 +1,543 @@
+        ids of namespaces this container should reside in such as {lldp: ["host", "0", "1"]}
+    """
+    container_list = []
+    containers_in_namespaces = defaultdict(list)
+
+    container_list_command = "redis-cli -n 4 keys \"FEATURE|*\""
+    command_output = duthost.shell(container_list_command)
+    exit_code = command_output["rc"]
+    pytest_assert(exit_code == 0, "Failed to get keys (container names) in `FEATURE` table")
+    for line in command_output["stdout_lines"]:
+        container_list.append(line.split("|")[1].strip())
+
+    for container_name in container_list:
+        if container_name in skip_containers:
+            continue
+
+        command_config_entry = "redis-cli -n 4 hgetall \"FEATURE|{}\"".format(container_name)
+        command_output = duthost.shell(command_config_entry)
+        exit_code = command_output["rc"]
+        pytest_assert(exit_code == 0, "Failed to get configuration of container '{}' in `FEATURE` table"
+                      .format(container_name))
+
+        is_enabled, has_global_scope, has_per_asic_scope = parse_config_entry(command_output["stdout_lines"])
+        if is_enabled != "disabled":
+            logger.info("Parsing the configuration of container '{}' in Config_DB.".format(container_name))
+            if num_asics > 1:
+                if has_global_scope == "True":
+                    containers_in_namespaces[container_name].append("host")
+                if has_per_asic_scope == "True":
+                    for asic_id in range(num_asics):
+                        containers_in_namespaces[container_name].append(str(asic_id))
+            else:
+                containers_in_namespaces[container_name].append("host")
+            logger.info("The configuration of container '{}' in `FEATURE` table was retrieved.".format(container_name))
+
+    return containers_in_namespaces
+
+
+def disable_containers_autorestart(duthost, containers_in_namespaces):
+    """Disables the autorestart of enabled containers.
+
+    Args:
+        duthost: Hostname of DUT.
+        containers_in_namespaces: A dictionary where keys are container names and
+        values are lists which contains ids of namespaces this container should reside in.
+
+    Returns:
+        None.
+    """
+    for container_name in containers_in_namespaces.keys():
+        logger.info("Disabling the autorestart of container '{}'.".format(container_name))
+        command_disable_autorestart = "sudo config feature autorestart {} disabled".format(container_name)
+        command_output = duthost.shell(command_disable_autorestart)
+        exit_code = command_output["rc"]
+        pytest_assert(exit_code == 0, "Failed to disable the autorestart of container '{}'".format(container_name))
+        logger.info("The autorestart of container '{}' was disabled.".format(container_name))
+
+
+def get_group_program_info(duthost, container_name, group_name):
```

Comment:
Should this function be a shared helper function?

Reply:
Moved this function to `tests/common/helers/dut_utils.py`.